waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -102,20 +108,38 @@ private void reconnectIfDisconnected() {
         throw new MetastoreUnavailableException("Client " + name + " is not available", e);
       }
     }
-
   }
 
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultMetaStoreClientFactory.class);
+
   /*
    * (non-Javadoc)
    * @see com.hotels.bdp.waggledance.client.MetaStoreClientFactoryI#newInstance(org.apache.hadoop.hive.conf.HiveConf,
    * java.lang.String, int)
    */
   @Override
   public CloseableThriftHiveMetastoreIface newInstance(HiveConf hiveConf, String name, int reconnectionRetries) {
-    ReconnectingMetastoreClientInvocationHandler reconnectingHandler = new ReconnectingMetastoreClientInvocationHandler(
-        hiveConf, name, reconnectionRetries);
-    return (CloseableThriftHiveMetastoreIface) Proxy.newProxyInstance(getClass().getClassLoader(), INTERFACES,
-        reconnectingHandler);
-  }
+    CloseableThriftHiveMetastoreIface closeableThriftHiveMetastoreIface = null;
+    final String closeableIFaceImpl = hiveConf.get(WaggleDanceHiveConfVars.CLOSEABLE_IFACE_IMPL.varname);
+    if (isNotEmpty(closeableIFaceImpl)) {
```

Comment:
Could do with some tests of this logic where we pass in a test class that is a bare bones client implementation and ensure this gets constructed properly. Then again but missing the required constructor to ensure the exception gets rethrown. Also pass in non-existing class name.